### PR TITLE
Get project id from route params and added checks

### LINKED ISSUE #4: bugfix: handle project not found
even if route is valid `/projects/id` in the eyes of braid, there still might not be a project with the given id. We should handle that gracefully

### PR DESCRIPTION
**Pull request by Kaivosukeltaja**
Fixes #4

### CODE REVIEW

File: src/Project.js [hunk 1]
```diff
@@ -68,12 +68,12 @@ class Project extends Component {
   state = { ...initialState };
 
   componentDidMount() {
-    this.fetchData(this.props.location.state.id);
+    this.fetchData(this.props.match.params.id);
```
[the remark this answers is not in the document]
Comment (Kaivosukeltaja) [reply src/Project.js]: I'd probably go with `match.params` myself - it gets populated if the browser has hit a URL with parameters, namely `/projects/:id`. I've never user `location.state` myself but I think it only contains anything if the route has been hit by a Redirect with the `state` prop? I did notice `location.state` was empty when accessing the project with a direct URL.

File: src/Navbar.js [hunk 2]
```diff
@@ -37,9 +37,11 @@ const Navbar = ({ name, projects, onLogout }) => (
                 render={props => (
                   <span>
                     {
-                      projects.find(
-                        prj => prj.project_id === props.location.state.id
-                      ).project_name
+                      typeof props.location.state !== 'undefined'
```
[the remark this answers is not in the document]
Comment (Kaivosukeltaja) [reply src/Navbar.js]: Possibly, also we should be checking if the `projects.find` returns anything before using its properties.

File: src/Project.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (deepredsky) [reply src/Project.js]: `match.params` sounds good to me, lets go with it 👍

[the remark this answers is not in the document]
Comment (jobn) [reply src/Project.js]: yeah, this seems like a misuse of `location.state` on my part. `location.state` is supposed to be used for forcing a route to render with a match that does not correspond to the current history location.
I'm 👍 for using `match.params`

File: src/Navbar.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jobn) [reply src/Navbar.js]: Another benefit of using `match.params` is that params is always present on match, whereas `state` is nullable on `location`. So we can remove the check for undefined.

**Comment by Kaivosukeltaja** [comment]
I rewrote the project selector text using `reduce()`, now it should work in all cases.

**Pull request merged**